backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java

```diff
@@ -47,7 +48,7 @@ public Element render() {
 
         String label = diagramDescription.getLabelProvider().apply(variableManager);
 
-        UUID diagramId = diagramDescription.getIdProvider().apply(variableManager);
+        UUID diagramId = optionalPreviousDiagram.map(Diagram::getId).orElseGet(() -> diagramDescription.getIdProvider().apply(variableManager));
```

Comment:
While I agree with this code, I don't think that it is necessary in THIS specific commit. It would be good with PCD's commit on stable identifiers but there are some consequences coming with such change. This will lead to the removal of both the GetOrCreateIdProvider and the deletion of the `idProvider` method entirely (which are both good things). If this change is not necessary, it may be better to postpone it to do it properly.

Reply:
It is not necessary here but it is necessary for nodes, so no need to remove it. It just brings something similar as PCD's PR but in a different way. Both will still be necessary, I'll write more about that in the ADR.